api/src/main/java/org/openmrs/BaseCustomizableData.java

```diff
@@ -9,124 +9,46 @@
  */
 package org.openmrs;
 
-import java.util.ArrayList;
 import java.util.Collection;
 import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Set;
 
 import org.openmrs.attribute.Attribute;
-import org.openmrs.customdatatype.CustomValueDescriptor;
 import org.openmrs.customdatatype.Customizable;
 
 /**
- * Extension of {@link BaseOpenmrsData} for classes that support customization via user-defined attributes.
+ * Extension of {@link BaseOpenmrsData} for classes that support customization via user-defined
+ * attributes.
+ * 
  * @param <A> the type of attribute held
  * @since 1.9
  */
 public abstract class BaseCustomizableData<A extends Attribute> extends BaseOpenmrsData implements Customizable<A> {
 	
-	private Set<A> attributes = new LinkedHashSet<A>();
+	private Collection<A> attributes = new LinkedHashSet<A>();
 	
 	/**
 	 * @see org.openmrs.customdatatype.Customizable#getAttributes()
 	 */
 	@Override
-	public Set<A> getAttributes() {
+	public Collection<A> getAttributes() {
```

Comment:
Since this is changing the signature of a public method, shouldn't we follow the guidelines at? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Deprecation